libs/vkd3d/state.c

```diff
@@ -2809,7 +2809,7 @@ static HRESULT compute_input_layout_offsets(const struct d3d12_device *device,
         else
             offsets[i] = input_slot_offsets[e->InputSlot];
 
-        input_slot_offsets[e->InputSlot] = align(offsets[i] + format->byte_count, 4);
+        input_slot_offsets[e->InputSlot] = align(offsets[i] + format->byte_count, min(4, format->byte_count));
```

Comment:
Nice find! Not sure this is a correct fix though. This aligns the next element based on the current one. I think a more correct formulation would be:

```
offsets[i] = align(offsets[i], min(4, format->byte_count));
input_slot_offsets[e->InputSlot] = offsets[i] + format->byte_count;
```

Reply:
Yeah, that makes more sense.

I guess ``input_slot_offsets`` needs to contain the unaligned offset, as there's no way to know how to align it in advance.

I've updated the patch with this fix.